src/tree/v2/ResourceGroupsTreeDataProvider.ts

```diff
@@ -78,8 +79,19 @@ export class ResourceGroupsTreeDataProvider extends vscode.Disposable implements
 
     onDidChangeTreeData: vscode.Event<void | ResourceGroupsItem | ResourceGroupsItem[] | null | undefined> = this.onDidChangeTreeDataEmitter.event;
 
-    getTreeItem(element: ResourceGroupsItem): vscode.TreeItem | Thenable<vscode.TreeItem> {
-        return element.getTreeItem();
+    async getTreeItem(element: ResourceGroupsItem): Promise<vscode.TreeItem> {
+
+        const contextValuesToAdd: string[] = [];
+        if (canViewProperties(element)) {
+            contextValuesToAdd.push('canViewProperties');
+        }
+
+        const ti = await element.getTreeItem();
+
+        ti.contextValue = [...(ti.contextValue?.split(';') ?? []), ...contextValuesToAdd].sort().join(';');
```

Comment:
Doesn't have to be in this PR, but we definitely need a class for consistently handling context values. I don't like having to do all kinds of sort/split/join/etc. all over the place.

@philliphoff this makes me wonder though; we maybe don't need the `quickPickOptions` field at all on `ResourceModelBase`. We are _already_ calling `getTreeItem` on each element (for presenting in the quick pick), which is going to give us whether or not it's a leaf node (via the `collapsibleState` field, same as what VSCode uses to decide if it's a leaf or not), and the context value as a string (in the `contextValue` field). `getTreeItem`'s implementation should multiplex over to the BDPs when needed, so we'll be getting all the context value(s) that the client extension wants.

From there, I think it's _reasonable_ to:
1. Split
2. Dedupe
3. Add additional stuff like `canViewProperties`
4. Sort
5. Join and give VSCode a single string

This carries the disadvantage that there's an implicit contract that `contextValue` is a delimiter-separated string. We could override VSCode's default `TreeItem` to have it be a string array, thus making it explicit, or just live with the implicit contract. What do you think?

Reply:
Also @alexweininger doing the above approach probably eliminates a lot of the compat code since, from the flow of the quick picker, everything is just a VSCode `TreeItem`.